android/uhabits-core/src/main/java/org/isoron/uhabits/core/io/HabitsCSVImporter.java

```diff
@@ -0,0 +1,160 @@
+package org.isoron.uhabits.core.io;
+
+import androidx.annotation.NonNull;
+
+import com.opencsv.CSVReader;
+
+import org.isoron.uhabits.core.models.Checkmark;
+import org.isoron.uhabits.core.models.Frequency;
+import org.isoron.uhabits.core.models.Habit;
+import org.isoron.uhabits.core.models.HabitList;
+import org.isoron.uhabits.core.models.ModelFactory;
+import org.isoron.uhabits.core.models.Timestamp;
+import org.isoron.uhabits.core.utils.ColorConstants;
+import org.isoron.uhabits.core.utils.DateFormats;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.text.ParseException;
+import java.util.Date;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import javax.inject.Inject;
+
+public class HabitsCSVImporter extends AbstractImporter {
+    private ModelFactory modelFactory;
+
+    @Inject
+    public HabitsCSVImporter(@NonNull HabitList habits,
+                             @NonNull ModelFactory modelFactory) {
```

Comment:
Also I think the formatting is a little bit weird. Is this according to the project guidelines (I'm not sure atm).